soroban-env-host/src/builtin_contracts/base_types.rs

```diff
@@ -469,6 +465,12 @@ impl Vec {
     }
 }
 
+macro_rules! vec_new {
```

Comment:
I think this should be reconciled with `host_vec![]` macro in test utils. I don't have a strong opinion on the final naming, but I would slightly prefer `[]` syntax to follow the SDK closer (which is for we generally tried to achieve with the builtin contracts). Though probably cleanup can wait for a followup PR to make this one simpler.

Reply:
done